### PR TITLE
Add an option to filter out preview aks releases when fetching versions

### PR DESCRIPTION
**Pull request by matthewi**

### CODE REVIEW

**Comment by matthewi** [comment]
```
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers	(cached) [no tests to run]
?   	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/client	[no test files]
=== RUN   TestAccDataSourceAzureRMKubernetesServiceVersions_basic
--- PASS: TestAccDataSourceAzureRMKubernetesServiceVersions_basic (24.44s)
=== RUN   TestAccDataSourceAzureRMKubernetesServiceVersions_filtered
--- PASS: TestAccDataSourceAzureRMKubernetesServiceVersions_filtered (24.68s)
=== RUN   TestAccDataSourceAzureRMKubernetesServiceVersions_nopreview
--- PASS: TestAccDataSourceAzureRMKubernetesServiceVersions_nopreview (25.27s)
PASS
```

**Comment by matthewi** [comment]
Rebased against the released version and ran a test locally:
```
$ terraform apply
data.azurerm_kubernetes_service_versions.preview: Refreshing state...
data.azurerm_kubernetes_service_versions.current: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

k8s_all_latest = 1.17.0
k8s_all_versions = [
  "1.13.11",
  "1.13.12",
  "1.14.7",
  "1.14.8",
  "1.15.5",
  "1.15.7",
  "1.16.1",
  "1.16.4",
  "1.17.0",
]
k8s_latest = 1.15.7
k8s_versions = [
  "1.13.11",
  "1.13.12",
  "1.14.7",
  "1.14.8",
  "1.15.5",
  "1.15.7",
]
```

File: azurerm/internal/services/containers/data_source_kubernetes_service_version.go [hunk 1]
```diff
@@ -87,6 +102,11 @@ func dataSourceArmKubernetesServiceVersionsRead(d *schema.ResourceData, meta int
 					continue
 				}
 
+				if (isPreview) && (allowPreview == false) {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/containers/data_source_kubernetes_service_version.go]: we can simplify this too:

```suggestion
				if isPreview && allowPreview == false {
```

File: azurerm/internal/services/containers/data_source_kubernetes_service_version.go [hunk 2]
```diff
@@ -40,6 +40,12 @@ func dataSourceArmKubernetesServiceVersions() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"allow_preview": {
```
Comment (tombuildsstuff) [inline azurerm/internal/services/containers/data_source_kubernetes_service_version.go]: I think this'd make more sense as:

```suggestion
			"include_preview": {
```

WDYT? Could we also [add this to the documentation](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/d/kubernetes_service_versions.html.markdown#L33) too?

Comment (matthewi) [reply azurerm/internal/services/containers/data_source_kubernetes_service_version.go]: Sure. I'll flip the default as well so it maintains the existing behavior by default.

File: website/docs/d/kubernetes_service_versions.html.markdown [hunk 3]
```diff
@@ -32,11 +32,14 @@ output "latest_version" {
...
 
+* `include_preview` - (Optional) Default: `true` Set to `false` to exclude preview
+versions of Kubernetes from the version results.
+
 ## Attributes Reference
 
 * `versions` - The list of all supported versions.
 
-* `latest_version` - The most recent version available.
+* `latest_version` - The most recent version available. If `include_preview == false`, this is the most recent non-preview version available.
```
Comment (tombuildsstuff) [inline website/docs/d/kubernetes_service_versions.html.markdown]: can we make this:

```suggestion
* `latest_version` - The most recent version available. If `include_preview == false`, this is the most recent non-preview version available.
```

File: website/docs/d/kubernetes_service_versions.html.markdown [hunk 4]
```diff
@@ -32,11 +32,14 @@ output "latest_version" {
 
 * `version_prefix` - (Optional) A prefix filter for the versions of Kubernetes which should be returned; for example `1.` will return `1.9` to `1.14`, whereas `1.12` will return `1.12.2`.
 
+* `include_preview` - (Optional) Default: `true` Set to `false` to exclude preview
```
Comment (tombuildsstuff) [inline website/docs/d/kubernetes_service_versions.html.markdown]: can we make this:

```suggestion
* `include_preview` - (Optional) Should Preview versions of Kubernetes in AKS be included? Defaults to `true`
```

To match other resources?

**Comment by matthewi** [comment]
lintrest and tflint pass on my machine. We'll see what Travis does this time 🤞

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-02-13 at 10 46 01

**Pull request merged**